x/crosschain/types/cctx.go

```diff
@@ -114,10 +115,21 @@ func (m *CrossChainTx) AddRevertOutbound(gasLimit uint64) error {
 		return fmt.Errorf("cannot revert before trying to process an outbound tx")
 	}
 
+	// in protocol contract V1, developers can specify a revert address for Bitcoin chains
+	// TODO: remove this V1 logic after switching Bitcoin to V2 architecture
+	// https://github.com/zeta-chain/node/issues/2711
+	revertReceiver := m.InboundParams.Sender
+	if m.ProtocolContractVersion == ProtocolContractVersion_V1 &&
+		chains.IsBitcoinChain(m.InboundParams.SenderChainId, []chains.Chain{}) {
+		revertAddress, valid := m.RevertOptions.GetBTCRevertAddress(m.InboundParams.SenderChainId)
```

Comment:
That is a good point though. The `sender` is the caller address (in EVM event) who initiated the inbound transaction. For EVM chain, it is [in here](https://github.com/zeta-chain/node/blob/fix-incorrect-zContext-origin/zetaclient/chains/evm/observer/v2_inbound.go#L186) ; for Solana chain, it is [in here](https://github.com/zeta-chain/node/blob/fix-incorrect-zContext-origin/zetaclient/chains/solana/observer/inbound.go#L206) . I think a invalid sender address only happens when >= 2/3 zetaclients post wrong sender address, which means a bug or being compromised.

When compromised, the hacker may directly move the funds from TSS to their own wallets. To prevent zetaclient bugs, it is reasonable to add extra code to make zetacore more defensive.